lnbits/wallets/alby.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+from typing import AsyncGenerator, Optional
+
+import httpx
+from loguru import logger
+
+from lnbits.settings import settings
+
+from .base import (
+    InvoiceResponse,
+    PaymentResponse,
+    PaymentStatus,
+    StatusResponse,
+    Unsupported,
+    Wallet,
+)
+
+
+class AlbyWallet(Wallet):
+    """https://guides.getalby.com/alby-wallet-api/reference/api-reference"""
+
+    def __init__(self):
+        endpoint = settings.alby_api_endpoint
+
+        if not endpoint or not settings.alby_api_key:
+            raise Exception("cannot initialize getalby")
+
+        self.endpoint = endpoint[:-1] if endpoint.endswith("/") else endpoint
+        self.auth = {"Authorization": "Bearer " + settings.alby_api_key}
+        self.client = httpx.AsyncClient(base_url=self.endpoint, headers=self.auth)
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("/balance", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        data = r.json()["balance"]
+        if r.is_error:
+            return StatusResponse(data["error"], 0)
+
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
+
+        r = await self.client.post(
+            "/invoices",
+            json={
+                "amount": amount,
+                "description": memo or "",
+            },
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()
+        checking_id = data["payment_hash"]
+        payment_request = data["payment_request"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        # https://api.getalby.com/payments/bolt11
+        r = await self.client.post(
+            "/payments/bolt11",
+            json={"invoice": bolt11},  # assume never need amount in body
```

Comment:
I think an amount needs to be passed to pay 0-amount invoices. I am not sure if this is something lnbits supports normally or not?

Reply:
The other existing wallet methods doesn't appear to have 0-amount as a parameter.
Maybe one of the LNBits team can comment on this better.

Reply:
@dni

Reply:
see base.py method interface definition: https://github.com/lnbits/lnbits/blob/3e91200d3f0b1d195be21309032bb04c3cbab297/lnbits/wallets/base.py#L76